### ISSUE TITLE
Compile without support for the embedded mode fails

### ISSUE THREAD

**Issue by Lo0k** [GrahamDumpleton/mod_wsgi#546]
Hi!

I try to compile `mod_wsgi` (version 4.7.1) Apache module, without support for the embedded mode, but it fails. If embedded mode is enabled, everything works fine.

Same for version 4.6.8.

On Debian 10, with the installed packages:
```
apache2-dev
python-all-dev
```
Configure step:
```
# ./configure --disable-embedded
checking for apxs2... /usr/bin/apxs2
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for prctl... yes
checking Apache version... 2.X
checking for python... /usr/bin/python
configure: creating ./config.status
config.status: creating Makefile
```
Then, make:
```
# make
/usr/bin/apxs2 -c -I/usr/include/python2.7 -DNDEBUG -D_FORTIFY_SOURCE=2 -DMOD_WSGI_DISABLE_EMBEDDED -Wc,-g -Wc,-O2  src/server/mod_wsgi.c src/server/wsgi_*.c -L/usr/lib -L/usr/lib/python2.7/config  -lpython2.7 -lpthread -ldl  -lutil -lm
/usr/share/apr-1.0/build/libtool  --mode=compile --tag=disable-static x86_64-linux-gnu-gcc -prefer-pic -pipe -g -O2 -fstack-protector-strong -Wformat -Werror=format-security  -Wdate-time -D_FORTIFY_SOURCE=2   -DLINUX -D_REENTRANT -D_GNU_SOURCE  -pthread  -I/usr/include/apache2  -I/usr/include/apr-1.0   -I/usr/include/apr-1.0 -I/usr/include -g -O2 -I/usr/include/python2.7 -DNDEBUG -D_FORTIFY_SOURCE=2 -DMOD_WSGI_DISABLE_EMBEDDED  -c -o src/server/mod_wsgi.lo src/server/mod_wsgi.c && touch src/server/mod_wsgi.slo
libtool: compile:  x86_64-linux-gnu-gcc -pipe -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -DLINUX -D_REENTRANT -D_GNU_SOURCE -pthread -I/usr/include/apache2 -I/usr/include/apr-1.0 -I/usr/include/apr-1.0 -I/usr/include -g -O2 -I/usr/include/python2.7 -DNDEBUG -D_FORTIFY_SOURCE=2 -DMOD_WSGI_DISABLE_EMBEDDED -c src/server/mod_wsgi.c  -fPIC -DPIC -o src/server/.libs/mod_wsgi.o
src/server/mod_wsgi.c: In function ‘wsgi_set_restrict_embedded’:
src/server/mod_wsgi.c:5046:13: error: ‘wsgi_python_required’ undeclared (first use in this function); did you mean ‘wsgi_reload_required’?
         if (wsgi_python_required == -1)
             ^~~~~~~~~~~~~~~~~~~~
             wsgi_reload_required
src/server/mod_wsgi.c:5046:13: note: each undeclared identifier is reported only once for each function it appears in
src/server/mod_wsgi.c: In function ‘wsgi_add_import_script’:
src/server/mod_wsgi.c:5338:9: error: ‘wsgi_python_required’ undeclared (first use in this function); did you mean ‘wsgi_reload_required’?
         wsgi_python_required = 1;
         ^~~~~~~~~~~~~~~~~~~~
         wsgi_reload_required
src/server/mod_wsgi.c: In function ‘wsgi_set_dispatch_script’:
src/server/mod_wsgi.c:5386:5: error: ‘wsgi_python_required’ undeclared (first use in this function); did you mean ‘wsgi_reload_required’?
     wsgi_python_required = 1;
     ^~~~~~~~~~~~~~~~~~~~
     wsgi_reload_required
src/server/mod_wsgi.c: In function ‘wsgi_set_access_script’:
src/server/mod_wsgi.c:5807:5: error: ‘wsgi_python_required’ undeclared (first use in this function); did you mean ‘wsgi_reload_required’?
     wsgi_python_required = 1;
     ^~~~~~~~~~~~~~~~~~~~
     wsgi_reload_required
src/server/mod_wsgi.c: In function ‘wsgi_set_auth_user_script’:
src/server/mod_wsgi.c:5847:5: error: ‘wsgi_python_required’ undeclared (first use in this function); did you mean ‘wsgi_reload_required’?
     wsgi_python_required = 1;
     ^~~~~~~~~~~~~~~~~~~~
     wsgi_reload_required
src/server/mod_wsgi.c: In function ‘wsgi_set_auth_group_script’:
src/server/mod_wsgi.c:5887:5: error: ‘wsgi_python_required’ undeclared (first use in this function); did you mean ‘wsgi_reload_required’?
     wsgi_python_required = 1;
     ^~~~~~~~~~~~~~~~~~~~
     wsgi_reload_required
src/server/mod_wsgi.c: In function ‘wsgi_hook_init’:
src/server/mod_wsgi.c:13337:9: error: ‘wsgi_python_required’ undeclared (first use in this function); did you mean ‘wsgi_reload_required’?
     if (wsgi_python_required == -1)
         ^~~~~~~~~~~~~~~~~~~~
         wsgi_reload_required
src/server/mod_wsgi.c: In function ‘wsgi_hook_child_init’:
src/server/mod_wsgi.c:13418:9: error: ‘wsgi_python_required’ undeclared (first use in this function); did you mean ‘wsgi_reload_required’?
     if (wsgi_python_required) {
         ^~~~~~~~~~~~~~~~~~~~
         wsgi_reload_required
apxs:Error: Command failed with rc=65536
.
make: *** [Makefile:31: src/server/mod_wsgi.la] Error 1
```

Any idea?

Thanks!

Tony

**Comment by Lo0k** [comment]
After a quick search, I found that the `wsgi_python_required` variable is [only declared](https://github.com/GrahamDumpleton/mod_wsgi/blob/develop/src/server/wsgi_interp.h#L63) if embedded mode is enabled at compile time.

However, this variable is unconditionally required by [wsgi_interp.c](https://github.com/GrahamDumpleton/mod_wsgi/blob/develop/src/server/wsgi_interp.c#L1995) and [mod_wsgi.c](https://github.com/GrahamDumpleton/mod_wsgi/blob/develop/src/server/mod_wsgi.c).

I'm not sure it is the right way, but [this commit](https://github.com/Lo0k/mod_wsgi/commit/ac3ec3597eb4c52dfd6535b1bf7e06e57407ec9c) resolves the issue.

Thanks,

Tony

**Comment by GrahamDumpleton** [comment]
Compiling without embedded mode probably has been tested for many years. Is there a specific reason you need to have it. You can instead add:

```
WSGIRestrictEmbedded On
```

to the Apache config after the ``LoadModule`` directive for the ``wsgi_module`` and has same effect but at run time.

**Comment by GrahamDumpleton** [comment]
Cleaning up old issues. The compilation problem will be fixed in mod_wsgi 4.9.0. Inclusion of embedded mode support is still enabled by default and when using ``mod_wsgi-express``, since 4.8.0, it is actually possible to enable embedded mode from command line. So long as your application is robust and doesn't need auto recovery support daemon mode supports, embedded can notably increase performance with the generated configuration ``mod_wsgi-express`` uses as is better tuned for Python applications than standard Apache installation.
